---
layout: default
title: Research
---

<div class="container">
    <div class="row">
        <div class="col-sm-10">
            <div class="page-header">
                <h3>Research</h3>
            </div>
        </div>
    </div>
</div>


<div class="container">
    <div class="row">
        <div class="col-sm-2">
        </div>
        <div class="col-sm-8">
            
            The <A HREF="/">CIBIV</A> wants to understand the
            processes that have shaped the genomes of contemporary species.
            To this end we apply methods from statistics, computer sciences,
            mathematics and computational statistics to develop models that
            mimic the process of evolution.
            These methods are further investigated in close collaboration with
            "wet" biologists to address real biological questions.
            
            <p>
            <P>Currently we are working (in collaboration with various colleagues) on
            the following aspects of molecular evolution:
            <br>
            
            <!--
             <ul>
             <li>Phylogenetic Tree Reconstruction
             <li>Efficient Algorithms to Analyze Large Datasets
             <li>Complex Evolutionary Models
             <li>Populations Genetics
             <li>Applying Parallel Computing to the Analysis of Large Dataset
             <li>Expression Data Analysis
             <li>Parameter Estimation for Evolutionary Models
             <li>Phylogenetic Inference with Missing Data
             </ul>
             -->
            <ul>
                <hr width="85%">
                <li><strong>Alignments</strong>
                    <br>Statistics of sequence alignment (i.e. mcmcalgn).
                    Recently we have extended this approach to reconstruct an alignment and
                    a phylogenetic tree simultaneously.
                </li>
                
                <hr width="85%">
                <li><strong>Sequence evolution</strong>
                    <br>To understand sequence evolution it is necessary to model the substitution
                    process. We are working on models sequence that allow dependencies among
                    sequence sites (Markov fields seem to be an appropriate tool).
                    We are developing test statistics to select the "best" model, to detect
                    groups of sequence that evolve differently form the rest of a gene family,
                    say. We have developed a test to detect change points (branches where the
                    substitution model changes) in a phylogenetic tree.
                    Currently we are working on methods to detect the dependency structure
                    among sequence positions in an alignment.
                </li>
                
                <hr width="85%">
                <li><strong>Gene trees</strong>
                    <br>We develop efficient heuristic algorithms to reconstruct trees based
                    on sequence data (i.e. <A HREF="http://www.tree-puzzle.de">TREE-PUZZLE</A>).
                    To this end we have developed parallel TREE-PUZZLE program.
                    Moreover, we are currently developing a variant of TREE-PUZZLE,
                    which computes (maximum) likelihood trees for up to 1,000 sequences
                    in reasonable time. We are also working on super tree methods to
                    merge different gene trees to form one species tree.
                    Quartet based tree reconstruction method appear as a versatile tool
                    to study super trees from a new perspective.
                </li>
                
                <hr width="85%">
                <li><strong>Population genetics</strong>
                    <br>Gene trees appear in a natural context also in populations, here, however,
                    the gene tree in a population is a random variable if a sample of sequences
                    is drawn from the population. We are interested in the development and
                    application of coalescence based methods to infer the demographic history
                    of populations. In the future we plan to work on coalescence processes with
                    complex interactions patterns. In this context we have constructed the so
                    called hvrbase, where currently most
                    of the hypervariable regions from the mitochondrial genome from primates
                    are collected in a multiple sequence alignment. This user friendly database
                    is currently extended to store other genomic regions.
                </li>
                
                <hr width="85%">
                <li><strong>Complex pattern of evolution</strong>
                    <br>To reconstruct the evolutionary history it is necessary to take more
                    complex events like lateral gene transfer (between species),
                    gene duplication, and gene loss into account. A combination of these
                    events may disturb the relation between species trees and gene trees.
                    Recently, we have developed a maximum likelihood based method to estimate
                    the amount of gene flow among prokaryotes by analyzing the COG database.
                    This full genome analysis poses a collection of new computational problems
                    as well as modeling problems. Our "Jukes Cantor" type of modeling gene
                    transfer needs refinements. Moreover, we have to take into account
                    duplication and losses of genes. This will be done in the next future.
                </li>
                
                <hr width="85%">
                <li><strong>Species tree</strong>
                <br>The topics outlined above will eventually be employed to reconstruct
                    one gigantic species tree utilizing all the sequence data available for
                    the different species. Models of sequence evolution are necessary to
                    detect differently evolving regions in complete genomes. Tree
                    reconstruction methods for a large number of sequences allow the
                    reconstruction of gene trees with several hundred sequences, and finally
                    the patchiness of the available sequence data for different species
                    makes it necessary to apply super tree methods. A better understanding 
                    of complex evolutionary patterns will also reveal instances where the gene 
                    trees are different from the species tree. Once this is well understood 
                    it seems reasonable to construct a sequenced based tree of life.
                </li>
            </ul>
        </div>
    </div>
    <div style="height:40px;"></div>
    
</div>









