---
layout: default
title: Software
---

<div class="container">
    <div class="row">
        <div class="col-sm-10">
            <div class="page-header">
                <h3>Software developed at the Department</h3>
            </div>
        </div>
    </div>
    
    <div class="row">
        <div style="height:20px;"></div>
        <div class="col-sm-6">
            <h4>Phylogenetics :: Inference</h4>
            <div class="jumbotron">
                <ul>
                    <li><b>IQ-TREE (Efficient reconstruction of large maximum-likelihood phylogenies)</b>
                        <!-- last changed/inserted 2015.08.26 -->
                        <br>An efficient maximum likelihood algorithm for phylogenetic tree reconstruction and an ultra-fast approximation for phylogenetic bootstrap.
                        The package is available at the
                        <a HREF="http://www.cibiv.at/software/iqtree">IQ-Tree homepage</a>.
                        <p>
                    </li>
                    
                    <li><b>TREE-PUZZLE</b>
                        <!-- last changed/inserted 2015.08.07 -->
                        <br>is a program package to reconstruct phylogenetic trees
                        from protein, DNA, and binary data by maximum likelihood.
                        The package is available at
                        <a HREF="http://www.tree-puzzle.de">www.tree-puzzle.de</a>.
                        <p>
                    </li>
                    
                    <li><b>IQPNNI (Important Quartet Puzzling with NNI Optimization)</b>
                        <!-- last changed/inserted 2013.03.06 -->
                        <br>A quartet based method to reconstruct phylogenetic trees
                        from really large datasets (DNA and protein).
                        The method applies a QP-like approach requiring only a
                        part of the quartets and some fast NNI optimization.
                        The package is available at the
                        <a HREF="http://www.cibiv.at/software/iqpnni">IQPNNI homepage</a>.
                        <p>
                        
                        <li><b>PDA (Phylogenetic Diversity Algorithm)</b>
                            <!-- last changed/inserted 2015.06.04 -->
                            <br>implements efficient algorithms to solve the Phylogenetic Diversity (PD) problem.
                            The package is available at the
                            <a HREF="http://www.cibiv.at/software/pda">PDA homepage</a>.
                            <p>
                        </li>
                        
                        <li><b>HGT</b>
                            <!-- last changed/inserted 2010.03.09 -->
                            <br>A software to simulate and estimate Horizontal Gene Transfer
                            events.
                            The package is available at the
                            <a HREF="http://www.cibiv.at/software/hgt">HGT homepage</a>.
                            <p>
                        </li>
                        
                        <li><b>TripleC</b>
                            <!-- last changed/inserted 2008.03.11 -->
                            <br>a program to constuct consensus trees from rooted triples.
                            The software is available at the
                            <a HREF="http://www.cibiv.at/software/triplec">TripleC homepage</a>.
                            <p>
                        </li>
                        </ul>
                <br>
                <br>
                <br>
                <br>
            </div>
        </div>
 
        <div class="col-sm-6">
            <h4>Next Generation Sequencing</h4>
            <div class="jumbotron">
                <ul>
                    <li><b>NextGenMap</b> (Next Generation Mapper)<br><!-- last changed/inserted 2013.05.17 -->
                        is a computer program to map NGS reads against a reference genome using CPUs or GPUs.
                        The program is available at the
                        <a HREF="http://www.cibiv.at/software/ngm/NGM">NextGenMap homepage</a>.
                        <p>
                    </li>
                    <li><b>Teaser</b>
                        <!-- last changed/inserted 2015.09.21 -->
                        <br>Teaser is a software package for optimizing NGS read mapping results.  Through rapid automated benchmarks on simulated or real data sets, the most suitable mapper for an experiment can be found within minutes. By testing the same mapper on multiple parameter settings, results can be further tuned for accuracy or speed. At the end of each benchmark, Teaser presents the results in a browser-viewable report that includes interactive figures.
                        
                        The software is available at the
                        <a HREF="http://teaser.cibiv.univie.ac.at/">Teaser homepage</a>.
                        <p>
                    </li>
                    <li><b>NGC</b>
                        <!-- last changed/inserted 2012.12.13 -->
                        <br>A compressor for high-throughput sequencing data.
                        The package is available at the
                        <a HREF="http://www.cibiv.at/~niko/ngc/">NGC homepage</a>.
                        <p>
                    </li>
                    <li><b>Bisulfite Sequencing Scrorer (BISS)</b>
                        <!-- last changed/inserted 2014.03.26 -->
                        <br>is an analysis pipeline for the deep sequencing data after
                        bisulfite conversion (BS-Seq).
                        The package is available at the
                        <a HREF="http://www.cibiv.at/software/ngm/BiSS">BiSS homepage</a>.
                        <p>
                    </li>
                    <li><b>MASon</b> (Million Alignments in Seconds)
                        <!-- last changed/inserted 2011.02.14 -->
                        <br>is a C++ library that uses high performance architectures to reduce the computing time of NGS analysis pipelines.
                        The program is available at the
                        <a HREF="http://www.cibiv.at/software/ngm/mason">MASon homepage</a>.
                        <p>
                    </li>
                    <li><b>SNP2GO</b>
                        <!-- last changed/inserted 2015.02.20 -->
                        <br>is an R package developed for the functional analysis of
                        Genome-Wide Association (GWA) studies. The package is available at the
                        <a HREF="http://www.cibiv.at/software/snp2go">SNP2GO homepage</a>.
                        <p>
                    </li>
                </ul>
            </div>
        </div>
        
    </div><!-- row -->
    
    <div style="height:30px;"></div>

    <div class="row">
        <div class="col-sm-6">
            <h4>Phylogenetics :: Modelling</h4>
            <div class="jumbotron">
                <ul>
                    <li><b>HaMSTR</b>
                        <!-- last changed/inserted 2014.10.28 -->
                        <br>HaMSTR is a Hidden Markov Model based search tool to screen EST sequence data for the presence of putative orthologs to a pre-defined set of genes.
                        HaMSTR is accessible at
                        <a HREF="http://www.deep-phylogeny.org/hamstr">www.deep-phylogeny.org/hamstr</a>.
                        <p>
                    </li>
                    <li><b>ImOSM (<EM>Im</EM>bed <EM>O</EM>ne <EM>S</EM>tep <EM>M</EM>utations into an alignment)</b>
                        <!-- last changed/inserted 2012.06.02 -->
                        <br>A program to Imbed model violation as One Step Mutations into a
                        sequence alignment. This is a useful tool to study the robustness
                        in phylogeny inference.
                        <a HREF="http://www.cibiv.at/software/imosm">ImOSM homepage</a>.
                        <p>
                    </li>
                    <li><b>misfits</b>
                        <!-- last changed/inserted 2011.06.09 -->
                        <br>is a computer program to evaluate the goodness of fit between a phylogenetic model and an alignment.
                        The program is available at the
                        <a HREF="http://www.cibiv.at/software/misfits">misfits homepage</a>.
                        <p>
                    </li>
                    <li><b>ALIFRITZ</b>
                        <!-- last changed/inserted 2012.11.16 -->
                        <br>performs statistical multiple alignment and phylogeny
                        reconstruction simultaneously.
                        The package is available at the
                        <a HREF="http://www.cibiv.at/software/alifritz">ALIFRITZ homepage</a>.
                        <p>
                    </li>
                    <li><b>SISSI (Simulating Sequence Evolution with Site-Specific Interactions)</b>
                        <!-- last changed/inserted 2010.03.09 -->
                        <br>SISSI is a software tool to  generate data  of related sequences
                        along a given phylogeny, taking into account user defined system of
                        neighbourhoods and instantaneous rate matrices.
                        The software is available at the
                        <a HREF="http://www.cibiv.at/software/sissi">SISSI homepage</a>.
                        <p>
                    </li>
                    <li><b>DupliDeli - Estimation of Gene Family Specific Duplication and Deletion Rates</b>
                        <!-- last changed/inserted 2014.11.17 -->
                        <br>is a program to estimate gene family specific duplication and deletion rates as well as the ancestral number of gene copies.
                        The package is available at the
                        <a HREF="http://www.cibiv.at/software/duplideli">DupliDeli homepage</a>.
                        <p>
                    </li>
                    <li><b>RecDetec - detecting recombination and phylogenetic information along alignments</b>
                        <!--13-04--><!-- last changed/inserted 2013.04.22 -->
                        RecDetec is a software to investigate recombination scenarios based on the bootscan principle along alignments of (viral) genomes. It also allowes to assess the content of phylogenetic information along alignments.
                        The package is available at the
                        <a HREF="http://www.cibiv.at/software/recdetec">RecDetec homepage</a>.
                        <p>
                    </li>
                    <li><b>REvolver: Modeling sequence evolution under domain constraints</b>
                        <!-- last changed/inserted 2012.11.12 -->
                        <br>REvolver is a program to simulate protein sequence evolution. REvolver automatically integrates domain information described by a profile Hidden Markov Model (pHMM) into the simulation.
                        The package is available at the
                        <a HREF="http://www.cibiv.at/software/revolver">REvolver homepage</a>.
                        <p>
                    </li>
                </ul>
            </div>
        </div>
        
        <div class="col-sm-6">
            <h4>Miscellaneous</h4>
            <div class="jumbotron">
                <ul>
                    <li><b>DAGwoman - DAG-based WOrkflow MANager</b>
                        <!-- last changed/inserted 2014.02.24 -->
                        <br>is a workflow engine that allows running and managing DAG-based scientific workflows on computing resources.
                        The package is available at the
                        <a HREF="http://www.cibiv.at/software/dagwoman">DAGwoman homepage</a>.
                        <p>
                    </li>
                    <li><b>EMOGEE (Estimator for MOdels of Gene Expression Evolution)</b>
                        <!-- last changed/inserted 2010.03.09 -->
                        <br>implements neutral models for evolution of gene expression to analyze or simulate microarray data.
                        EMOGEE and related tools are available at the
                        <a HREF="http://www.cibiv.at/software/emogee">EMOGEE homepage</a>.
                        <p>
                    </li>
                    <li><b>Epi-Speller</b>
                        <!-- last changed/inserted 2013.02.19 -->
                        <br>is a program for analyzing multiple genome-wide profiling epigenomic data.
                        The package is available at the <a HREF="http://www.cibiv.at/software/epispeller">Epi-Speller homepage</a>
                        <p>
                    </li>
                    <li><b>GeoMeTree (Geodesic Metric on Trees)</b>
                        <!-- last changed/inserted 2010.03.09 -->
                        <br>GeoMeTree is a python program to compute the geodesic path between two phylogenetic trees.
                        The package is available at the
                        <a HREF="http://www.cibiv.at/software/geometree">GeoMeTree homepage</a>.
                        <p>
                    </li>
                    <li><b>NGV (Next Generation Viewer)</b>
                        <!-- last changed/inserted 2010.00.00 -->
                        <br>A Preprocessor and Browser for efficient display of large HTS Data Sets.
                        The package is available at the
                        <a HREF="http://purl.org/lsdv/core ">NGV homepage</a>.
                        <p>
                    </li>
                    <li><b>TAXNAMECONVERT</b>
                        <!-- last changed/inserted 2010.03.09 -->
                        <br>is a program to rename sequence and taxon names in tree and sequence files.
                        The software is available at the
                        <a HREF="http://www.cibiv.at/software/taxnameconvert">Taxnameconvert homepage</a>.
                        <p>
                    </li>
                    <li><b>TreeSnatcher</b>
                        <!-- last changed/inserted 2010.06.29 -->
                        <br>is a GUI-driven Java application that recognizes bifurcating
                        and multifurcating trees in pixel images (like JPG)
                        and outputs the tree in Newick format.
                        The software is available at the
                        <a HREF="http://www.cibiv.at/software/treesnatcher">TreeSnatcher homepage</a>.
                        <p>
                    </li>
                    <li>
                        <a href="/software/rest">More</a>
                    </li>
                </ul>
            </div>
        </div>
    </div>
    <div style="height:40px;"></div>
</div>


    
    
    
    
    
    
    
        